src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -120,6 +122,28 @@ public ReceivedMessage withHook(InteractionHook hook)
         return this;
     }
 
+    private void checkIntent()
+    {
+        // Checks whether access to content is limited and the message content intent is not enabled
+        if (api != null && !contentIntentWarning && !api.isIntent(GatewayIntent.MESSAGE_CONTENT))
+        {
+            SelfUser selfUser = api.getSelfUser();
+            if (!Objects.equals(selfUser, author) && !mentions.getUsers().contains(selfUser) && isFromGuild())
```

Comment:
Something I'm curious about here:
What about messages from the REST api? Don't they also require the Intent to be turned on in the admin dashboard?

So it could be that we received a message via REST and the dashboard didn't have the intent turned on but we used it in the gateway, so this warning wouldn't show, right?

Or does discord not let a WS connection establish with MESSAGE_CONTENT if it isn't enabled for REST too in the dashboard?

Reply:
I would hope it can only be enabled in the gateway if its enabled in the application dashboard, similar to all other privileged intents.

Reply:
Works for me.